dallinger/command_line.py

```diff
@@ -501,74 +472,63 @@ def deploy_sandbox_shared_setup(verbose=True, app=None, web_procs=1, exp_config=
         "whimsical": config["whimsical"],
     }
 
-    for key in heroku_config:
-        subprocess.check_call([
-            "heroku",
-            "config:set",
-            "{}={}".format(key, quote(str(heroku_config[key]))),
-            "--app", app_name(id)
-        ], stdout=out)
+    for k, v in heroku_config.items():
+        heroku_app.set(k, v)
 
     # Wait for Redis database to be ready.
     log("Waiting for Redis...")
     ready = False
     while not ready:
-        redis_url = subprocess.check_output([
-            "heroku", "config:get", "REDIS_URL", "--app", app_name(id),
-        ])
-        r = redis.from_url(redis_url)
+        r = redis.from_url(heroku_app.redis_url)
         try:
             r.set("foo", "bar")
             ready = True
         except redis.exceptions.ConnectionError:
             time.sleep(2)
 
     log("Saving the URL of the postgres database...")
-    subprocess.check_call(["heroku", "pg:wait", "--app", app_name(id)])
-    db_url = subprocess.check_output([
-        "heroku", "config:get", "DATABASE_URL", "--app", app_name(id)
-    ])
-
     # Set the notification URL and database URL in the config file.
     config.extend({
-        "notification_url": u"http://" + app_name(id) + ".herokuapp.com/notifications",
-        "database_url": db_url.rstrip().decode('utf8'),
+        "notification_url": heroku_app.url + "/notifications",
+        "database_url": heroku_app.db_url,
```

Comment:
For `dallinger sandbox --verbose`. I receive a TypeError. I just casted it to unicode and fixed it. I'll push after making sure there aren't anymore TypeErrors.

❯❯ Saving the URL of the postgres database...
Waiting for database postgresql-elliptical-25308... available
Traceback (most recent call last):
  File "/Users/vlall/.virtualenvs/d2/bin/dallinger", line 11, in <module>
    load_entry_point('dallinger', 'console_scripts', 'dallinger')()
  File "/Users/vlall/.virtualenvs/d2/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/vlall/.virtualenvs/d2/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/vlall/.virtualenvs/d2/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/vlall/.virtualenvs/d2/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/vlall/.virtualenvs/d2/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/vlall/prog/dev/Dallinger/dallinger/command_line.py", line 557, in sandbox
    _deploy_in_mode(u'sandbox', app, verbose)
  File "/Users/vlall/prog/dev/Dallinger/dallinger/command_line.py", line 549, in _deploy_in_mode
    deploy_sandbox_shared_setup(verbose=verbose, app=app)
  File "/Users/vlall/prog/dev/Dallinger/dallinger/command_line.py", line 493, in deploy_sandbox_shared_setup
    "database_url": heroku_app.db_url,
  File "/Users/vlall/prog/dev/Dallinger/dallinger/config.py", line 126, in extend
    expected_type=expected_type,
TypeError: Got 'https://dlgr-5187084d.herokuapp.com//notifications' for notification_url, expected <type 'unicode'>

Reply:
@vlall Just checked my integration test, and that fails for the same reason. I think the properties and return values from HerokuApp methods should probably be changed to return unicode objects.